flutter-studio/src/io/flutter/module/FlutterPackageStep.java

```diff
@@ -9,51 +9,86 @@
 import com.android.tools.idea.wizard.model.ModelWizardStep;
 import com.android.tools.idea.wizard.model.SkippableWizardStep;
-import com.intellij.ide.projectWizard.ModuleNameLocationComponent;
-import com.intellij.ide.util.projectWizard.NamePathComponent;
-import com.intellij.ide.util.projectWizard.WizardContext;
+import com.intellij.ide.IdeBundle;
+import com.intellij.ide.highlighter.ModuleFileType;
+import com.intellij.ide.util.BrowseFilesListener;
+import com.intellij.ide.util.projectWizard.*;
 import com.intellij.openapi.Disposable;
 import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
 import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectBundle;
+import com.intellij.openapi.roots.ui.configuration.ProjectStructureConfigurable;
+import com.intellij.openapi.ui.Messages;
 import com.intellij.openapi.ui.TextComponentAccessor;
+import com.intellij.openapi.ui.TextFieldWithBrowseButton;
+import com.intellij.openapi.util.Comparing;
 import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.ui.ComboboxWithBrowseButton;
+import com.intellij.ui.DocumentAdapter;
 import io.flutter.FlutterBundle;
 import io.flutter.sdk.FlutterSdkUtil;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import javax.swing.*;
+import javax.swing.event.DocumentEvent;
 import java.io.File;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Optional;
 
+@SuppressWarnings("Duplicates")
 public class FlutterPackageStep extends SkippableWizardStep<FlutterModuleModel> implements Disposable {
 
   @NotNull private final BindingsManager myBindings = new BindingsManager();
   @NotNull private final ListenerManager myListeners = new ListenerManager();
   @NotNull private final ValidatorPanel myValidatorPanel;
 
+  private final WizardContext myWizardContext;
+
   @SuppressWarnings("deprecation") // TODO(messick): Update the root panel with whatever the latest and greatest is.
   @NotNull
   private final StudioWizardStepPanel myRootPanel;
   private final ModuleValidator myModuleValidator;
   private FlutterModuleBuilder myBuilder;
   private JPanel myPanel;
   private ComboboxWithBrowseButton myFlutterSdkPath;
-  private ModuleNameLocationComponent myModuleNameLocationComponent;
+  private JPanel myModulePanel;
+  private JTextField myModuleName;
+  private TextFieldWithBrowseButton myModuleContentRoot;
+  private TextFieldWithBrowseButton myModuleFileLocation;
+  private boolean myModuleNameChangedByUser = false;
+  private boolean myModuleNameDocListenerEnabled = true;
+  private boolean myContentRootChangedByUser = false;
+  private boolean myContentRootDocListenerEnabled = true;
+  private boolean myImlLocationChangedByUser = false;
```

Comment:
Are these for implementing the 'advanced' section of the wizard page? We need to inline them because the form definition is in a different module?

We could also choose to not have an advanced section - most (all?) users won't use it, and that's a lot of code we then don't have to maintain.